### PR TITLE
Make Template::Resolver always cache

### PR DESCRIPTION
**Pull request by jhawthorn**
All actionview caches are already cleared at the start of each request (when Resolver.caching is false) by `PerExecutionDigestCacheExpiry`, which calls `LookupContext::DetailsKey.clear`, which clears all caches. (These methods/classes could use a rename as their purpose is more than just digest cache related, but I'll save that for a future PR)

Because caches are always cleared per-request in dev, we shouldn't need this extra logic to compare mtimes and conditionally reload templates. I've tested this in a local development app. @tenderlove also confirms this should be `rm`- able.

This should make templates faster in development when rendering the same one multiple times.

This also enables the possibility of only clearing the template cache when files change, which I'll attempt in a future PR.

### CODE REVIEW

**Comment by kaspth** [comment]
I’ve wanted to delete this code for years! ❤️🤘

File: actionview/lib/action_view/template.rb [hunk 1]
```diff
@@ -122,10 +122,10 @@ def self.finalize_compiled_template_methods=(_)
 
     extend Template::Handlers
 
-    attr_reader :source, :identifier, :handler, :original_encoding, :updated_at
+    attr_reader :source, :identifier, :handler, :original_encoding
```
Comment (kamipo) [inline actionview/lib/action_view/template.rb]: `:update_at` had exist for about 9 years. 38d78f99d52801d8392a7229b40edae74cc3d142
Should it be deprecated first like `:original_encoding`? 0e13941dbe9c4840175e5c6f5b2c3f572249a4ec

Comment (rafaelfranca) [reply actionview/lib/action_view/template.rb]: Yes, we should.

Comment (jhawthorn) [reply actionview/lib/action_view/template.rb]: on it!

**Pull request merged**